src/VisualStudio/Core/Def/DocumentOutline/DocumentOutlineControl_ModelComputation.cs

```diff
@@ -0,0 +1,257 @@
+                ThreadingContext.ThrowIfNotOnUIThread();
+                if (EditorAdaptersFactoryService.GetBufferAdapter(textBuffer) is IPersistFileFormat persistFileFormat &&
+                    ErrorHandler.Succeeded(persistFileFormat.GetCurFile(out var filePath, out _)))
+                {
+                    return filePath;
+                }
+
+                return null;
+            }
+        }
+
+        /// <summary>
+        /// Starts a new task to update the UI model.
+        /// </summary>
+        private void StartUpdateUIModelTask()
+        {
+            // 'true' value is unused.  this just signals to the queue that we have work to do.
+            _updateUIModelQueue.AddWork(true);
+        }
+
+        /// <summary>
+        /// Filters and sorts the UI model.
+        /// </summary>
+        private async ValueTask<DocumentSymbolModel?> UpdateUIAsync(ImmutableSegmentedList<bool> unused, CancellationToken cancellationToken)
+        {
+            var model = await _computeUIModelQueue.WaitUntilCurrentBatchCompletesAsync().ConfigureAwait(false);
+            if (model is null)
+                return null;
+
+            // Switch to the UI thread to get the current search query and latest active text view.
+            await ThreadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
+
+            var searchQuery = searchBox.Text;
+
+            var activeTextView = GetLastActiveIWpfTextView();
+            if (activeTextView is null)
+                return null;
+
+            // Switch to the threadpool to filter and sort the DocumentSymbolItems.
+            await TaskScheduler.Default;
+
+            var updatedDocumentSymbolItems = model.DocumentSymbolItems;
+
+            if (!string.IsNullOrWhiteSpace(searchQuery))
+                updatedDocumentSymbolItems = DocumentOutlineHelper.Search(updatedDocumentSymbolItems, searchQuery, cancellationToken);
+
+            updatedDocumentSymbolItems = DocumentOutlineHelper.Sort(updatedDocumentSymbolItems, SortOption, cancellationToken);
+
+            StartHightlightNodeTask(ExpansionOption.NoChange);
+
+            return new DocumentSymbolModel(updatedDocumentSymbolItems, model.OriginalSnapshot);
+        }
+
+        /// <summary>
+        /// Starts a new task to highlight the symbol node corresponding to the current caret position in the editor, expand/collapse
+        /// nodes (if applicable), and updates the UI.
+        /// </summary>
+        private void StartHightlightNodeTask(ExpansionOption expansionOption)
+        {
+            _highlightAndExpandNodesQueue.AddWork(expansionOption);
+        }
+
+        /// <summary>
+        /// Highlights the symbol node corresponding to the current caret position in the editor, expands/collapses nodes, then updates the UI.
+        /// </summary>
+        private async ValueTask HightlightNodeAsync(ImmutableSegmentedList<ExpansionOption> expansionOption, CancellationToken cancellationToken)
```

Comment:
this method does more than just highlight afaict, it also sets the actual *UI* items on the last line.  It would be good if *this* method indicated that it has responsibility for taking the final computed model information and making the UI show it.  For example, it could be named `DetermineHighlightedItemAndPresentItems` or something like that.  `Present` is a good work to indicate we are actually now finally showing the user something.